slatedb/src/oracle.rs

```diff
@@ -16,19 +20,29 @@ pub(crate) struct Oracle {
     /// The sequence number of the most recent write that has been fully durable
     /// flushed to the remote storage.
     pub(crate) last_remote_persisted_seq: Arc<MonotonicSeq>,
+    /// Tracks the mapping from sequence number to timestamp.
+    pub(crate) seq_tracker: RwLock<TieredSequenceTracker>,
```

Comment:
Probably makes sense to just use a `std::sync::Mutex`. The only heavy-weight read being done is the manifest flush and that's always a single task. Lookups into the structure should be super fast.